presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -197,12 +196,37 @@ private ResourceGroupInfo getSummaryInfo()
                     runningTimeLimit,
                     queuedTimeLimit,
                     DataSize.succinctBytes(cachedMemoryUsageBytes),
-                    queuedQueries.size() + descendantQueuedQueries,
-                    runningQueries.size() + descendantRunningQueries,
+                    getQueuedQueries(),
+                    getRunningQueries(),
                     eligibleSubGroups.size(),
                     subGroups.values().stream()
-                            .map(InternalResourceGroup::getSummaryInfo)
-                            .collect(toImmutableList()));
+                            .filter(group -> group.getRunningQueries() + group.getQueuedQueries() > 0)
```

Comment:
I don't think it's the missing of filter that exploded data, it's calling `getSummaryInfo` for all child groups. The next line should be the actual fix.

Reply:
Yes, I went beyond the 'actual' fix and re-thought what data needs to be in here.